featuretools/entityset/entity.py

```diff
@@ -249,7 +250,7 @@ def query_by_values(self, instance_vals, variable_id=None, columns=None,
             df = self.df.head(0)
 
         else:
-            if isinstance(instance_vals, dd.Series):
+            if isinstance(instance_vals, dd.Series) or isinstance(instance_vals, ks.Series):
                 df = self.df.merge(instance_vals.to_frame(), how="inner", on=variable_id)
             else:
                 df = self.df[self.df[variable_id].isin(instance_vals)]
```

Comment:
In testing on the instacart dataset for the predict next purchase problem, this code block appears to be causing a performance issue.